File: app/src/main/java/za/org/grassroot2/util/NetworkUtil.java
package za.org.grassroot2.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;


final public class NetworkUtil {

    public static boolean hasInternetAccess(Context context) {
        boolean hasInternet = false;
        ConnectivityManager cm = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = cm.getActiveNetworkInfo();
        if (networkInfo != null) {
            hasInternet = networkInfo.isConnectedOrConnecting();
        }
        return hasInternet;
    }
}

File: app/src/main/java/za/org/grassroot2/util/NetworkUtil.kt
package za.org.grassroot2.util

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo


object NetworkUtil {

    fun hasInternetAccess(context: Context): Boolean {
        var hasInternet = false
        val cm = context
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = cm.activeNetworkInfo
        if (networkInfo != null) {
            hasInternet = networkInfo.isConnectedOrConnecting
        }
        return hasInternet
    }
}
